rero_ils/modules/notifications/dispatcher.py

```diff
@@ -44,178 +44,23 @@ def dispatch_notifications(cls, notification_pids=None, resend=False,
-                            item_data.get('pid'))
-                        # get the requested loan it can be in several states
-                        # due to the automatic request validation
-                        request_loan = None
-                        for state in [
-                            LoanState.ITEM_AT_DESK,
-                            LoanState.ITEM_IN_TRANSIT_FOR_PICKUP,
-                            LoanState.PENDING
-                        ]:
-                            request_loan = item.get_first_loan_by_state(state)
-                            if request_loan:
-                                break
-                        request_patron = Patron.get_record_by_pid(
-                            request_loan['patron_pid'])
-                        ctx_data['request_patron'] = \
-                            request_patron.replace_refs().dumps()
-                        pickup_location = Location.get_record_by_pid(
-                            request_loan['pickup_location_pid'])
-                        ctx_data['request_pickup_name'] = \
-                            pickup_location['pickup_name']
-                    documents_data['item'] = {
-                        'barcode': item_data['barcode']
-                    }
-                    call_number = item_data.get('call_number')
-                    if call_number:
-                        documents_data['item']['call_number'] = call_number
-                    second_call_number = item_data.get('second_call_number')
-                    if second_call_number:
-                        documents_data['item']['second_call_number'] = \
-                            second_call_number
-                    location = item_data.get('location')
-                    if location:
-                        loc = Location.get_record_by_pid(location.get('pid'))
-                        email = loc.get('notification_email')
-                        if email:
-                            ctx_data['location_email'] = email
-                        documents_data['item']['location_name'] =\
-                            loc.get('name')
-                        library = loc.get_library()
-                        documents_data['item']['library_name'] =\
-                            library.get('name')
-
-                aggregated[n_type][l_pid][p_pid]['documents'].append(
-                    documents_data
-                )
-                aggregated[n_type][l_pid][p_pid]['notifications'].append(
-                    notification
-                )
+            try:
+                notification = Notification.get_record_by_pid(notification_pid)
+                if notification:
+                    dispatcher_function = cls._process_notification(
+                        notification, resend, aggregated)
+            except Exception as e:
+                current_app.logger.error(
+                    f'Notification has not be sent (pid: {notification_pid}): '
```

Comment:
It would be nice to have the loan pid and type of notification in the error log.